src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -669,8 +660,12 @@ private static void ValidateCacheKey(object key)
         /// </summary>
         private sealed class CoherentState
         {
+#if NETCOREAPP
+            private readonly ConcurrentDictionary<string, CacheEntry> _stringEntries = [];
```

Comment:
with this change, we probably don't need `StringKeyComparer` to exist at all in netcoreapp